platform/src/components/aws/bucket.ts

```diff
@@ -385,16 +430,29 @@ export interface BucketArgs {
    */
   cors?: Input<false | Prettify<BucketCorsArgs>>;
   /**
-   * Enable versioning for the bucket.
-   *
-   * Bucket versioning enables you to store multiple versions of an object, protecting
-   * against accidental deletion or overwriting.
-   *
-   * @default `false`
+   * The lifecycle configuration for the bucket.
    * @example
+   * This example will delete every object in the bucket after 30 days.
    * ```js
    * {
-   *   versioning: true
+   *  lifecycle: [{
+   *   enabled: true,
+   *   expiresIn: "30 days"
+   *  }]
+   * }
+   * ```
+   */
+  lifecycle?: Input<Input<Prettify<BucketLifecycleArgs>>[]>;
+  /**
+   * Delete objects in the "/tmp" directory after 30 days.
+   * ```js
+   * {
+   *   lifecycle: [
+   *     {
+   *       prefix: "/tmp",
+   *       expiresIn: "30 days"
+   *     }
+   *   ]
    * }
    * ```
    */
```

Comment:
something weird happened were this was placed below the property instead of above

```suggestion
   * The lifecycle configuration for the bucket.
   * @example
   * Delete objects in the "/tmp" directory after 30 days.
   * ```js
   * {
   *   lifecycle: [
   *     {
   *       prefix: "/tmp",
   *       expiresIn: "30 days"
   *     }
   *   ]
  * }
  * ```
  */
  lifecycle?: Input<Input<Prettify<BucketLifecycleArgs>>[]>;
```